### DISCUSSION TITLE
Error in table pageHistory after backup/restore

### DISCUSSION THREAD

**Discussion by lordslair** [requarks/wiki#5754] [Help / Questions]
Hello everyone

I installed previously WikiJS on a MariaDb server. 
Due to some changes, I had to migrate to a MySQL server

I used a MariaDB dump I already have, and imported it into MySQL.
During import, I had errors on `pageHistory.extra` field and the default value
MySQL doesn't allow that: 
`#1101 - BLOB, TEXT, GEOMETRY or JSON column 'extra' can't have a default value`

So I ALTERed the table to be able to do the import, and I thought I could be fine like that.
But, no (lol).

I suppose it's due to differences from your driver `mariadb` I used in the past which allowed this
And now, as I'm using the `mysql` driver with a MySQL server, it can't work like that anymore

Do you have any hints on what I can do here ?
I searched in the previous threads and didn't find any related questions
I added in attachment my actual config. I can provide more details if needed

TL;DR: How to properly migrate from MariaDB > MySQL, and dealing with `pageHistory.extra` default value

```
Docker: requarks/wiki:2.5
OS: Alpine Linux 3.16
Wiki.js version: 2.5,289
Database engine: MySQL 8.0
```

![Capture d’écran 2022-10-17 à 15 09 03](https://user-images.githubusercontent.com/5469390/196185731-a7b3abfc-4f54-4849-bc22-8a2e4d81e06d.png)

**Comment by calaad** [comment]
Same situation here, on debian 11. 
Migrate from mariadb to mysql8 : 2 tables ("pageHistory" and "pages") have default value what mysql8 does not accept.

**Comment by calaad** [comment]
@lordslair 
Find a solution/workaround : just remove the not null constraint
In your wikijs database : 
`alter table pageHistory modify extra longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin CHECK (json_valid(`extra`));`
`alter table pages modify extra longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin CHECK (json_valid(`extra`));`

It seems to do the job.
